### PR TITLE
Await for a Bridge to listen to its port

### LINKED ISSUE #192: Bridge.run() spawns a floating promise for listening to a port
https://github.com/matrix-org/matrix-appservice-bridge/blob/8c85b240441b742f191707098e74d8a10b1ed1a8/src/bridge.js#L377

We should await this function as it returns a promise that we may need to handle.

### PR DESCRIPTION
**Pull request by jaller94**
Fixes #192 

This allows for errors of `.listen` to bubble up and get handled.

### CODE REVIEW

**Comment by jaller94** [comment]
I will need to return a Bluebird promise for our tests.
Also, this may have increased the time we have to wait beyond Jasmine's default timeout.

**Comment by Half-Shot** [comment]
How come? This shouldn't affect the tests?

**Comment by jaller94** [comment]
Some tests use `returnValue.done()` instead of `.then()` as the last handler, which only works if `.run()` returns a Bluebird-Promise, but not a native Promise.

**Comment by Half-Shot** [comment]
oh, fair enough then.

File: src/bridge.js [hunk 1]
```diff
@@ -374,8 +375,11 @@ Bridge.prototype.run = function(port, config, appServiceInstance, hostname) {
         this._metrics.addAppServicePath(this);
     }
 
-    this.appService.listen(port, hostname);
-    return this.loadDatabases();
+    // We MUST return a Bluebird-Promise instead of a Promise.
+    // promise.done() is used by many tests in this repo.
+    return this.loadDatabases().then(async() => {
```
Comment (Half-Shot) [inline src/bridge.js:380]: Stupid question, why don't we use await here?

Comment (jaller94) [reply src/bridge.js:380]: Great question.
`loadDatabase` must be executed immediately to initialize the stores needed by some promises above. These promises will run as soon as `run` is done. The only other way is to move `loadDatabase` above the usage of `Promise.cast()`&`this._onAliasQuery()` and await it there.

`loadDatabase` returns a Bluebird-Promise, which is exactly what I need to return to make the tests happy. I just chain `.listen` in a native promise at the end, but the return value will stay a Bluebird-Promise.

Comment (Half-Shot) [reply src/bridge.js:380]: Fair enough, what a mess though :S

**Pull request merged**